packages/core/src/utils/lowercase.ts

```diff
@@ -0,0 +1,6 @@
+/**
+ * Transforms the input string to lower case.
+ */
+export function toLowerCase<T extends string>(value: T) {
+  return value.toLowerCase() as T;
```

Comment:
There's a cool generic in typescript called Lowercase that you could use here. 

```ts
return value.toLowerCase() as Lowercase<T>;
```

Reply:
Just pushed a commit that uses `Lowercase<T>` and updates those spots to use the util - thanks!